### DISCUSSION TITLE
How to run UI application using actions on windows self hosted runner

### DISCUSSION THREAD

**Discussion by sagartrivedi11** [community/community#67003] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have self-hosted windows runner where I'm running the workflow to start the notepad.
GitHub connection is started as `service `in windows runner. When I execute the workflow unable to see notepad UI in desktop but can see that process in task manager. While making the GitHub connection by running `run.cmd` and starting the same workflow I can see notepad UI on desktop.

**Hit & try till now in GH service:**
- changed the service Logon account as current user.
- Tried by giving the `local system` account as well allowed desktop interaction in service

Is there any way to launch the UI application if GH connection is active from service ?

**Comment by romanliv** [comment]
Feels you cannot do exactly this without workarounds. I would propose not to connect this runner as a service, but open Task Scheduler and create there a task that calls `run.cmd` when user logs-in. You will be able to run GUI applications from the runners connected like that.
